sdk/formrecognizer/azure-formrecognizer-perf/src/main/java/com/azure/formrecognizer/perf/CustomModelRecognitionTest.java

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.formrecognizer.perf;
+
+import com.azure.ai.formrecognizer.models.RecognizedForm;
+import com.azure.formrecognizer.perf.core.ServiceTest;
+import com.azure.perf.test.core.PerfStressOptions;
+import reactor.core.publisher.Mono;
+
+import java.util.List;
+
+/**
+ * Performs custom model recognition operations.
+ */
+public class CustomModelRecognitionTest extends ServiceTest<PerfStressOptions> {
+
+    private static final String URL_TEST_FILE_FORMAT = "https://raw.githubusercontent.com/Azure/azure-sdk-for-java/"
+        + "master/sdk/formrecognizer/azure-ai-formrecognizer/src/test/resources/sample_files/Test/";
+    private static final String FORM_JPG = "Form_1.jpg";
+
+    /**
+     * The CustomModelRecognitionTest class.
+     *
+     * @param options the configurable options for perf testing this class
+     */
+    public CustomModelRecognitionTest(PerfStressOptions options) {
+        super(options);
+    }
+
+    @Override
+    public Mono<Void> globalSetupAsync() {
+        return super.globalSetupAsync();
+    }
+
+    @Override
+    public Mono<Void> globalCleanupAsync() {
+        return super.globalCleanupAsync();
+    }
+
+    @Override
+    public void run() {
+        List<RecognizedForm> recognizedForms =
+            formrecognizerClient.beginRecognizeCustomFormsFromUrl(modelId, URL_TEST_FILE_FORMAT + FORM_JPG)
+                .getFinalResult();
+        assert recognizedForms != null;
+    }
+
+    @Override
+    public Mono<Void> runAsync() {
+        return formrecognizerAsyncClient
+            .beginRecognizeCustomFormsFromUrl(modelId, URL_TEST_FILE_FORMAT + FORM_JPG)
+            .count()
+            .flatMap(count -> count > 0
+                ? Mono.empty()
```

Comment:
We should ensure we wait till the operation finishes, and then wrap that in Mono.

Reply:
updated